src/deluge/gui/ui/keyboard_screen.cpp

```diff
@@ -383,11 +409,21 @@ int KeyboardScreen::buttonAction(int x, int y, bool on, bool inCardRoutine) {
 
 void KeyboardScreen::selectEncoderAction(int8_t offset) {
 	InstrumentClipMinder::selectEncoderAction(offset);
+	instrumentClipView.scrollVertical(0, 0, false);
+	uiNeedsRendering(this, 0xFFFFFFFF, 0);
 }
 
 int KeyboardScreen::getNoteCodeFromCoords(int x, int y) {
-	InstrumentClip* clip = getCurrentClip();
-	return clip->yScrollKeyboardScreen + x + y * clip->keyboardRowInterval;
+
+	Instrument* instrument = (Instrument*)currentSong->currentClip->output;
+	if (instrument->type == INSTRUMENT_TYPE_KIT) { //
+
+		return 60 + (int)(x / 4) + (int)(y / 4) * 4;
```

Comment:
maybe for future work, but should this be changeable? I know MIDI drum machines have all kinds of whacky interfaces (channel/sound, note/sound but not programmable, etc) so maybe it's best left for followup.

Reply:
This is solely internal to the deluge (each row in a kit gets a note code, same as each row in a synth)

You can still have the drum in this row be a midi drum with whatever note and channel you'd like, and map any external controller to it